src/stdlib_linalg_repmat.fypp

```diff
@@ -0,0 +1,30 @@
+#:include "common.fypp"
+#:set RCI_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES + INT_KINDS_TYPES
+submodule (stdlib_linalg) stdlib_linalg_repmat
+
+  implicit none
+
+contains
+
+  #:for k1, t1 in RCI_KINDS_TYPES
+    pure module function repmat_${t1[0]}$${k1}$(a, m, n) result(res)
+      ${t1}$, intent(in) :: a(:,:)
+      integer, intent(in) :: m,n
+      ${t1}$ :: res(m*size(a,1),n*size(a,2))
+      integer ::i,j,k,l
+      associate(ma=>size(a,1),na=>size(a,2))
+         do j=1,n
```

Comment:
@fortran-lang/stdlib what's our latest consensus on the use of `do concurrent` in stdlib? The 4-nested loop can be written more concisely with `do concurrent`, but I recall an argument against this is that it could interfere if `repmat` were called from within an OpenMP directive. Is `do concurrent` in stdlib still discouraged?

Reply:
gfortran will do loop-interchange in `do concurrent`, But it may not the best order.

Reply:
OK, let's leave it as is then.

Reply:
I can't retrieve the discussion, neither remember what was decided. I believe @awvwgk was also involve in it.